ui_testing/testcases/basic_tests/test04_testunits.py

```diff
@@ -1018,6 +1018,92 @@ def test025_edit_approach_overview_button(self):
+        Allow user to change between the two options specification and limit of quantification from edit mode.
+        """
+
+        testunits_request = self.test_unit_api.get_all_test_units(filter='{"typeName":2}').json()
+        self.assertEqual(testunits_request['status'], 1)
+        testunits = testunits_request['testUnits']
+        self.assertNotEqual(len(testunits), 0)
+        testunit_name = ''
+        for testunit in testunits:
+            if specification_type == 'spec':
+                if testunit['specifications'] != '' and testunit['quantification'] == '':
+                    testunit_name = testunit['name']
+                    break
+            elif specification_type == 'quan':
+                if testunit['specifications'] == '' and testunit['quantification'] != '':
+                    testunit_name = testunit['name']
+                    break
+            elif specification_type == 'spec_quan':
+                if testunit['specifications'] != '' and testunit['quantification'] != '':
+                    testunit_name = testunit['name']
+                    break
+
+        testunit_record = self.test_unit_page.search(value=testunit_name)[0]
+
+        self.test_unit_page.open_edit_page(row=testunit_record)
+        
+        self.base_selenium.LOGGER.info('generate random lower/ upper limit')
+        random_lower_limit = self.test_unit_page.generate_random_number(lower=0, upper=49)
+        random_upper_limit = self.test_unit_page.generate_random_number(lower=50, upper=100)
+        
+        if specification_type == 'spec':
+            self.base_selenium.LOGGER.info('switch to quantification')
+            self.test_unit_page.switch_from_spec_to_quan(lower_limit=random_lower_limit, upper_limit=random_upper_limit)
+            self.base_selenium.LOGGER.info('refresh to make sure that data are updated successfully')
+            self.base_selenium.refresh()
+
+            self.assertEqual(self.test_unit_page.get_testunit_specification_type(), 'quan')
+            self.assertEqual(self.test_unit_page.get_quan_upper_limit(), str(random_upper_limit))
+            self.assertEqual(self.test_unit_page.get_quan_lower_limit(), str(random_lower_limit))
+
+        elif specification_type == 'quan':
+            self.base_selenium.LOGGER.info('switch to specification')
+            self.test_unit_page.switch_from_quan_to_spec(lower_limit=random_lower_limit, upper_limit=random_upper_limit)
+            self.base_selenium.LOGGER.info('refresh to make sure that data are updated successfully')
+            self.base_selenium.refresh()
+
+            self.assertEqual(self.test_unit_page.get_testunit_specification_type(), 'spec')
+            self.assertEqual(self.test_unit_page.get_spec_upper_limit(), str(random_upper_limit))
+            self.assertEqual(self.test_unit_page.get_spec_lower_limit(), str(random_lower_limit))
+    
```

Comment:
'spec_quan' isn't implemented

Reply:
yep, it is removed as it is not specified in the testcase